diff --git a/dist/KeyValue/KeyValueField.js b/dist/KeyValue/KeyValueField.js
index 98e30835c94d3c13fe226edab40feae0f4d1f420..ac66f94b6348c47f8e838d04216bbad37767f74f 100644
--- a/dist/KeyValue/KeyValueField.js
+++ b/dist/KeyValue/KeyValueField.js
@@ -3,17 +3,27 @@ import { forwardRef, useImperativeHandle, useRef } from 'react';
 import { useSuggestions } from '../hooks/suggestions';
 const STRING_SCHEMA = { type: 'string' };
+const getDisplayValue = (value) => {
+    if (value === null || value === undefined) {
+        return '';
+    }
+    if (typeof value === 'object') {
+        return JSON.stringify(value);
+    }
+    return value;
+};
 export const KeyValueField = forwardRef(({ id, name, placeholder, value, 'data-testid': dataTestId, onChange, onFocus, onBlur }, ref) => {
     const inputRef = useRef(null);
+    const displayValue = getDisplayValue(value);
     useImperativeHandle(ref, () => inputRef.current);
     const { suggestionsMenu } = useSuggestions({
         propName: name,
         schema: STRING_SCHEMA,
         inputRef,
-        value: value !== null && value !== void 0 ? value : '',
+        value: displayValue,
         setValue: onChange,
     });
     return (_jsxs(TextInputGroup, { children: [_jsx(TextInputGroupMain, { type: "text", id: id, name: name, "data-testid": dataTestId, onChange: (_event, value) => {
                     onChange === null || onChange === void 0 ? void 0 : onChange(value);
-                }, ref: inputRef, onFocus: onFocus, onBlur: onBlur, placeholder: placeholder, value: value }), suggestionsMenu] }));
+                }, ref: inputRef, onFocus: onFocus, onBlur: onBlur, placeholder: placeholder, value: displayValue }), suggestionsMenu] }));
 });
 //# sourceMappingURL=KeyValueField.js.map
