diff --git a/dist/esm/index.js b/dist/esm/index.js
index 6e4d05f..101170e 100644
--- a/dist/esm/index.js
+++ b/dist/esm/index.js
@@ -924,18 +924,6 @@ function FlatpickrInstance(element, instanceConfig) {
                 return elem.contains(eventTarget_1);
             });
             if (lostFocus && isIgnored) {
-                if (self.config.allowInput) {
-                    self.setDate(self._input.value, false, self.config.altInput
-                        ? self.config.altFormat
-                        : self.config.dateFormat);
-                }
-                if (self.timeContainer !== undefined &&
-                    self.minuteElement !== undefined &&
-                    self.hourElement !== undefined &&
-                    self.input.value !== "" &&
-                    self.input.value !== undefined) {
-                    updateTime();
-                }
                 self.close();
                 if (self.config &&
                     self.config.mode === "range" &&
